Tests/PackageLoading/ConventionTests.swift

```diff
@@ -72,6 +98,295 @@ class ConventionTests: XCTestCase {
+
+        let name = "pkg"
+        // FIXME: This fails currently, it is a bug.
+        #if false
+        PackageBuilderTester(name, in: fs) { result in
+            result.checkModule(name) { moduleResult in
+                moduleResult.check(c99name: name, type: .executable, isTest: false)
+                moduleResult.checkSources(root: "/", paths: "main.swift", "Bar.swift")
+            }
+        }
+        #endif
+
+        fs = InMemoryFileSystem()
+        try fs.createDirectory(AbsolutePath("/hello.swift"))
+        try fs.createEmptyFiles("/Sources/main.swift",
+                                "/Sources/Bar.swift")
+
+        PackageBuilderTester(name, in: fs) { result in
+            result.checkModule(name) { moduleResult in
+                moduleResult.check(c99name: name, type: .executable, isTest: false)
+                moduleResult.checkSources(root: "/Sources", paths: "main.swift", "Bar.swift")
+            }
+        }
+
+        fs = InMemoryFileSystem()
+        try fs.createDirectory(AbsolutePath("/Sources/exe/hello.swift"), recursive: true)
+        try fs.createEmptyFiles("/Sources/exe/main.swift",
+                                "/Sources/exe/Bar.swift")
+
+        PackageBuilderTester(name, in: fs) { result in
+            result.checkModule("exe") { moduleResult in
+                moduleResult.check(c99name: "exe", type: .executable, isTest: false)
+                moduleResult.checkSources(root: "/Sources/exe", paths: "main.swift", "Bar.swift")
+            }
+        }
+    }
+
+    func testMultipleSwiftModules() throws {
+        var fs = InMemoryFileSystem()
+        try fs.createEmptyFiles("/Sources/A/main.swift",
+                                "/Sources/A/foo.swift",
+                                "/Sources/B/main.swift",
+                                "/Sources/C/Foo.swift")
+
+        PackageBuilderTester("MultipleModules", in: fs) { result in
+            result.checkModule("A") { moduleResult in
+                moduleResult.check(c99name: "A", type: .executable, isTest: false)
+                moduleResult.checkSources(root: "/Sources/A", paths: "main.swift", "foo.swift")
+            }
+
+            result.checkModule("B") { moduleResult in
+                moduleResult.check(c99name: "B", type: .executable, isTest: false)
+                moduleResult.checkSources(root: "/Sources/B", paths: "main.swift")
+            }
+
+            result.checkModule("C") { moduleResult in
+                moduleResult.check(c99name: "C", type: .library, isTest: false)
+                moduleResult.checkSources(root: "/Sources/C", paths: "Foo.swift")
+            }
+        }
+    }
+
+    func testMultipleClangModules() throws {
+        var fs = InMemoryFileSystem()
+        try fs.createEmptyFiles("/Sources/A/main.c",
+                                "/Sources/A/foo.h",
+                                "/Sources/A/foo.c",
+                                "/Sources/B/include/foo.h",
```

Comment:
headers are not picked up by SwiftPM at this stage, only during modulemap generation which is `Build`